src/FsAutoComplete.Core/Utils.fs

```diff
@@ -822,3 +831,91 @@ type FSharpSymbol with
     | :? FSharpStaticParameter as s -> s.DeclarationLocation
     | :? FSharpParameter as p -> p.DeclarationLocation
     | _ -> failwith $"cannot fetch DefinitionRange for unknown FSharpSymbol subtype {x.GetType().FullName}"
+
+module Tracing =
+
+  open System.Diagnostics
+  open FsOpenTelemetry
+  open StreamJsonRpc
+  open System.Collections.Generic
+
+  module SemanticConventions =
+    /// <remarks>
+    /// <see href="https://github.com/dotnet/fsharp/blob/c68285b00e2f53607d836aa0a11ab21abd556842/src/Compiler/Utilities/Activity.fs#L13"> From F# compiler</see>
+    /// </remarks>
+    module FCS =
+      [<Literal>]
+      let fileName = "fileName"
+
+      [<Literal>]
+      let project = "project"
+
+      [<Literal>]
+      let qualifiedNameOfFile = "qualifiedNameOfFile"
+
+      [<Literal>]
+      let userOpName = "userOpName"
+
+    [<Literal>]
+    let fsac_sourceCodePath = "fsac.sourceCodePath"
+
+    [<Literal>]
+    let projectFilePath = "fsac.projectFilePath"
+
+  [<Literal>]
+  let serviceName = "FsAutoComplete"
+
+  /// <remarks>
+  /// <see href="https://github.com/dotnet/fsharp/blob/c68285b00e2f53607d836aa0a11ab21abd556842/src/Compiler/Utilities/Activity.fs#L43"> From F# compiler</see>
+  /// </remarks>
+  [<Literal>]
+  let fscServiceName = "fsc"
+
+  let fsacActivitySource = new ActivitySource(serviceName, Version.info().Version)
+
+  /// <summary>
+  /// StreamJsonRpcTracingStrategy participates in and propagates trace context in  vs-streamjsonrpc
+  /// </summary>
+  ///
+  /// <remarks>
+  /// <see href="https://github.com/microsoft/vs-streamjsonrpc/blob/main/doc/tracecontext.md"> vs-streamjsonrpc tracecontext documentation</see>
+  /// </remarks>
+  type StreamJsonRpcTracingStrategy(activitySource: ActivitySource) =
+    interface IActivityTracingStrategy with
+      member this.ApplyInboundActivity(request: Protocol.JsonRpcRequest) : IDisposable =
+        let tags =
+          [
+            "rpc.system", box "jsonrpc"
+            "rpc.jsonrpc.is_notification", box request.IsNotification
+            "rpc.jsonrpc.is_response_expected", box request.IsResponseExpected
+            "rpc.jsonrpc.version", box request.Version
+            "rpc.jsonrpc.request_id", box request.RequestId
+            "rpc.method", box request.Method
```

Comment:
Some of these `rpc.` [come from here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md).  I could move the ones specified by OTel in the FsOpenTelemetry lib.